Kaldi-Python install problem concerning : /kaldi/src/util/kaldi-table-inl.h: and has no member named 'ExtractRange'

Dear Dmitry

While I am trying to install kaldi-python, I run into the following error message:

xiaofeng@spears:~/tool/kaldi-python$ python setup.py install
running install
running build
running build_py
make -C kaldi_io depend
make[1]: Entering directory '/home/xiaofeng/tool/kaldi-python/kaldi_io'
g++ -I/home/xiaofeng/tool/kaldi/src -I/home/xiaofeng/anaconda2/include/python2.7 -I/home/xiaofeng/anaconda2/include/python2.7 -I/home/xiaofeng/anaconda2/lib/python2.7/site-packages/numpy/core/include -M -msse -msse2 -Wall -I.. -pthread -DKALDI_DOUBLEPRECISION=0 -DHAVE_POSIX_MEMALIGN -Wno-sign-compare -Wno-unused-local-typedefs -Winit-self -DHAVE_EXECINFO_H=1 -rdynamic -DHAVE_CXXABI_H -DHAVE_ATLAS -I/home/xiaofeng/tool/kaldi/tools/ATLAS/include -I/home/xiaofeng/tool/kaldi/tools/openfst/include -Wno-sign-compare -g -fPIC .cpp > .depend.mk

make[1]: Leaving directory '/home/xiaofeng/tool/kaldi-python/kaldi_io'
make -C kaldi_io
make[1]: Entering directory '/home/xiaofeng/tool/kaldi-python/kaldi_io'
g++ -shared -o kaldi_io_internal.so -Wall -fPIC -I/home/xiaofeng/tool/kaldi/src -I/home/xiaofeng/anaconda2/include/python2.7 -I/home/xiaofeng/anaconda2/include/python2.7 -I/home/xiaofeng/anaconda2/lib/python2.7/site-packages/numpy/core/include -msse -msse2 -Wall -I.. -pthread -DKALDI_DOUBLEPRECISION=0 -DHAVE_POSIX_MEMALIGN -Wno-sign-compare -Wno-unused-local-typedefs -Winit-self -DHAVE_EXECINFO_H=1 -rdynamic -DHAVE_CXXABI_H -DHAVE_ATLAS -I/home/xiaofeng/tool/kaldi/tools/ATLAS/include -I/home/xiaofeng/tool/kaldi/tools/openfst/include -Wno-sign-compare -g -fPIC kaldi_io_internal.cpp /home/xiaofeng/tool/kaldi/src/lm/kaldi-lm.a /home/xiaofeng/tool/kaldi/src/decoder/kaldi-decoder.a /home/xiaofeng/tool/kaldi/src/lat/kaldi-lat.a /home/xiaofeng/tool/kaldi/src/hmm/kaldi-hmm.a /home/xiaofeng/tool/kaldi/src/transform/kaldi-transform.a /home/xiaofeng/tool/kaldi/src/gmm/kaldi-gmm.a /home/xiaofeng/tool/kaldi/src/tree/kaldi-tree.a /home/xiaofeng/tool/kaldi/src/matrix/kaldi-matrix.a /home/xiaofeng/tool/kaldi/src/util/kaldi-util.a /home/xiaofeng/tool/kaldi/src/base/kaldi-base.a /home/xiaofeng/tool/kaldi/src/thread/kaldi-thread.a -rdynamic -Wl,-rpath=/home/xiaofeng/tool/kaldi/tools/openfst/lib -L-lpython2.7 -lpthread -ldl -lutil -lm -L/home/xiaofeng/tool/kaldi/tools/openfst/lib -lfst /usr/lib/libatlas.so.3 /usr/lib/libf77blas.so.3 /usr/lib/libcblas.so.3 /usr/lib/liblapack_atlas.so.3 -lm -lpthread -ldl -lpython2.7 -lboost_python -lboost_system 
In file included from /home/xiaofeng/anaconda2/lib/python2.7/site-packages/numpy/core/include/numpy/ndarraytypes.h:1781:0,
from /home/xiaofeng/anaconda2/lib/python2.7/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
from /home/xiaofeng/anaconda2/lib/python2.7/site-packages/numpy/core/include/numpy/arrayobject.h:4,
from kaldi_io_internal.cpp:10:
/home/xiaofeng/anaconda2/lib/python2.7/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
#warning "Using deprecated NumPy API, disable it by " \
^
In file included from /home/xiaofeng/tool/kaldi/src/util/kaldi-table.h:467:0,
from kaldi_io_internal.cpp:21:
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >' has no member named 'ExtractRange'
if (range_holder_.ExtractRange(holder_, range)) {
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PyObjectHolder; std::string = std::basic_string]':
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1813:9: error: 'class PyObjectHolder' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >' has no member named 'ExtractRange'
if (!range_holder_.ExtractRange(holder_, range_)) {
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PythonToKaldiHolder >; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PythonToKaldiHolder >' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In instantiation of 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PyObjectHolder]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:175:29: required from 'const T& kaldi::SequentialTableReaderScriptImpl::Value() [with Holder = PyObjectHolder; kaldi::SequentialTableReaderScriptImpl::T = boost::python::api::object]'
kaldi_io_internal.cpp:482:1: required from here
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:351:9: error: 'class PyObjectHolder' has no member named 'ExtractRange'
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PyObjectHolder; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PyObjectHolder]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder, kaldi::BasicVectorHolder, kaldi::BasicVectorHolder > >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >, kaldi::BasicVectorVectorHolder, kaldi::BasicVectorVectorHolder > >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::RandomAccessTableReaderScriptImpl::HasKeyInternal(const string&, bool) [with Holder = PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >; std::string = std::basic_string]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:1825:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h: In member function 'bool kaldi::SequentialTableReaderScriptImpl::EnsureObjectLoaded() [with Holder = PythonToKaldiHolder >, kaldi::BasicPairVectorHolder, kaldi::BasicPairVectorHolder > >]':
/home/xiaofeng/tool/kaldi/src/util/kaldi-table-inl.h:360:3: warning: control reaches end of non-void function [-Wreturn-type]
}
^
In file included from /home/xiaofeng/tool/kaldi/src/base/kaldi-common.h:38:0,
from /home/xiaofeng/tool/kaldi/src/util/kaldi-io.h:30,
from kaldi_io_internal.cpp:20:
/home/xiaofeng/tool/kaldi/src/base/kaldi-math.h: At global scope:
/home/xiaofeng/tool/kaldi/src/base/kaldi-math.h:130:17: warning: 'kaldi::kLogZeroBaseFloat' defined but not used [-Wunused-variable]
const BaseFloat kLogZeroBaseFloat = -std::numeric_limits::infinity();
^
Makefile:31: recipe for target 'kaldi_io_internal.so' failed
make[1]: ** [kaldi_io_internal.so] Error 1
make[1]: Leaving directory '/home/xiaofeng/tool/kaldi-python/kaldi_io'
Makefile:10: recipe for target 'all' failed
make: *** [all] Error 2
running build_scripts
running install_lib
running install_scripts
changing mode of /home/xiaofeng/anaconda2/bin/apply-global-cmvn.py to 755
changing mode of /home/xiaofeng/anaconda2/bin/copy-feats-padded.py to 755
changing mode of /home/xiaofeng/anaconda2/bin/show-wav-ali.py to 755
changing mode of /home/xiaofeng/anaconda2/bin/compute-global-cmvn-stats.py to 755
running install_egg_info
Removing /home/xiaofeng/anaconda2/lib/python2.7/site-packages/kaldi_python-1.0-py2.7.egg-info
Writing /home/xiaofeng/anaconda2/lib/python2.7/site-packages/kaldi_python-1.0-py2.7.egg-info

I am sure that I installed the newest kaldi, could that be the cause of the this problem? Does kaldi-python requires older verison of kaldi?
Best

Many thanks

--- a/kaldi_io/Makefile
+++ b/kaldi_io/Makefile
@@ -21,7 +21,7 @@ TESTFILES =
 
 PYLIB = $(shell python-config --libs)
 PYINC = $(shell python-config --includes)
-NPINC = -I$(shell python -c 'import numpy; print numpy.get_include()')
+NPINC = -I$(shell python -c 'import numpy; print(numpy.get_include())')
 
 PYLIBS = kaldi_io_internal.so
 

--- a/kaldi_io/kaldi_io_internal.cpp
+++ b/kaldi_io/kaldi_io_internal.cpp
@@ -116,6 +116,11 @@ class PyObjectHolder {
     std::swap(t_, another->t_);
   }
 
+  bool ExtractRange(const PyObjectHolder &other, const std::string &range) {
+      KALDI_ERR << "ExtractRange is not defined for this type of holder.";
+      return false;
+  }
+
   ~PyObjectHolder() {}
 
 private:
@@ -259,6 +264,11 @@ class PythonToKaldiHolder {
 
   void Swap(PythonToKaldiHolder<Converter> * another) {
     std::swap(t_, another->t_);
+  }
+
+  bool ExtractRange(const PythonToKaldiHolder<Converter> &other, const std::string &range) {
+      KALDI_ERR << "ExtractRange is not defined for this type of holder.";
+      return false;
   }
 
   ~PythonToKaldiHolder() {}

